src/pages/blog/performance-update.md

```diff
@@ -0,0 +1,57 @@
+---
+templateKey: blog-post
+title: Performance Update
+author: Julia Kreger
+date: 2021-07-09T22:00:00.000Z
+category:
+  - label: Development Update
+    id: category-A7fnZYrE9
+---
+
+## What a Journey ##
+
+When I started out on a journey of trying to improve the performance of ironic a few months ago, the journey ended up taking a bit of a different path from what I expected. But I can now say with confidence, becasue all of the patches have merged and even backported to our "Wallaby" release, that we've made Ironic faster.
+
+The path was similar to what was expected, but half of the effort became building consensus and sharing context. Plus, very verbose notes in code *do* tend to help. Along this path, we also learned some new aspects, and were able to incorporate changes to hopefully improve the overall situation for most operators.
+
+### Where can I get this?! ###
+
+* Expect this in Ironic 18.1 or Ironic 19 (Xena) (whichever ends up being the next release).
+* Alternatively, Ironic 17.0.4 (Wallaby) since we backported most of the changes, and you can manually add database indexes if you need them.
+
+If you need some of these patches backported to older releases, join on us irc.oftc.net in #openstack-ironic and let us know.
+
+### Where did we end up? ###
+
+Overall, we have been able to reduce the time it took to download our sample data by 82% and improve the number of nodes returned per second to consumers like Nova who request lists of nodes with distinct fields in excess of 630% from our very first performance test measurements of Ironic while in the process of exploring performance bottlenecks and how to address them.
+
+This whole effort challenged how we approached operations with node lists.
+<br>
+* Additional database indexes
+* Improvements to how we sanitize nodes
+* Improvements to how we handle the access control context when lists of nodes are being handled.
+* Removal of excess work from the Database, and Database client
+<br>
+
+![Graph of database and api nodes returned per second via the Ironic API"](/img/blog-performance-update.png)
```

Comment:
This graph, in context of this blog post, only makes sense IMO if we demonstrate the old numbers with the new.

Reply:
That would be ironic 17.0 in the graph.

Reply:
Explicitly stressing 17.0 was just before we started this work.